Task/http/users/http.go

```diff
@@ -0,0 +1,147 @@
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"net/http"
+	"strconv"
+
+	"zopsmart/Task/models"
+	service "zopsmart/Task/services"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/gorilla/mux"
+)
+
+type UserHandler struct {
+	s service.Services
+}
+
+func New(service service.Services) UserHandler {
+	return UserHandler{s: service}
+}
+
+func (u UserHandler) AllUserDetails(w http.ResponseWriter,r *http.Request) {
+
+	w.Header().Set("content-type","application/json")
+
+	data, err := u.s.GetAllUsersService()
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+	resp, _ := json.Marshal(data)
+	w.WriteHeader(http.StatusOK)
+	w.Write(resp)
+}
+
+func (u UserHandler) GetUserById(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("content-type", "application/json")
+	id, err := strconv.Atoi(mux.Vars(r)["id"])
+	if err != nil {
+		resErr, _ := w.Write([]byte("invalid Id"))
+		w.WriteHeader(http.StatusBadRequest)
+		b,_ := json.Marshal(resErr)
+		w.Write(b)
+		return
+	}
+
+	res, err := u.s.GetUserByIdService(id)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		b,_ := json.Marshal(res)
+		w.Write(b)
+		return
+	} 
+
+	b, _ := json.Marshal(res)
```

Comment:
we will return a output like {data: user}

Reply:
yeah sir, will output in user form only, GetUserByIdService will return user